src/promptflow/promptflow/_core/token_provider.py

```diff
@@ -41,7 +41,21 @@ def _init_instance(self):
             # Initialize a credential instance
             from azure.identity import DefaultAzureCredential
 
-            self.credential = DefaultAzureCredential()
+            # Support sovereign cloud cases, like mooncake, fairfax.
```

Comment:
can we just return DefaultAzureCredential if azure-ai-ml is not installed, instead of throw exception here ?

Reply:
Agree, and may add info/warning message to hint user install azure-ai-ml if they want to use sovereign region.